addons/survey/controllers/main.py

```diff
@@ -251,11 +252,17 @@ def submit(self, survey, **post):
     def print_survey(self, survey, token=None, **post):
         '''Display an survey in printable view; if <token> is set, it will
         grab the answers of the user_input_id that has <token>.'''
+        UserInput = request.env['survey.user_input']
+        user_input = UserInput.sudo().search([('token', '=', token)], limit=1)
```

Comment:
Here you don't have to put user_input in its own variable doing ```user_input = request.env['survey.user_inpu'].sudo().search([('token', '=', token)], limit=1)``` 
is perfectly fine